src/Stratis.Bitcoin.Features.Wallet/WalletTransactionHandler.cs

```diff
@@ -346,7 +350,8 @@ protected void AddCoins(TransactionBuildContext context)
                 // then it's safe to stop adding UTXOs to the coin list.
                 // The primary goal is to reduce the time it takes to build a trx
                 // when the wallet is bloated with UTXOs.
-                if (sum > totalToSend)
+                // If we are only adding selected inputs then add them all.
+                if (sum > totalToSend && !onlySelectedInputs)
```

Comment:
If `onlySelectedInputs` is false and we do allow additional inputs, not all selected inputs will be chosen, which is a problem.
I think the selected inputs need to always be added to the list of coins, then we check the sum.

Reply:
That means the only change required would be to first add the selected inputs unconditionally.

Reply:
Nailed it.

Reply:
I've made a simple change that should do the trick unless we want to retain the order of selected inputs.